service/frontend/namespace_handler.go

```diff
@@ -778,6 +786,25 @@ func (d *namespaceHandlerImpl) mergeNamespaceData(
 	return old
 }
 
+func (d *namespaceHandlerImpl) upsertCustomSearchAttributesAliases(
+	current map[string]string,
+	upsert map[string]string,
+) (map[string]string, error) {
+	result := util.CloneMapNonNil(current)
+	for key, value := range upsert {
+		if value == "" {
+			delete(result, key)
+		} else if _, ok := current[key]; !ok {
+			result[key] = value
+		} else {
+			return nil, serviceerror.NewInvalidArgument(
```

Comment:
hmm ask user to retry invalid argument looks weird. 
Currently if there're concurrent namespace update, persistence layer will return Unavailable, which will be retried and customer most likely won't see the error.

Reply:
But this Unavailable error retry is in the grpc level, which won't work because the args will be the same from the operator handler.
The retry from the user will make operator handler to generate a new map to be upsert based on newer version of the namespace data.

Reply:
If user must change agrs, then `InvalidArgument` is the right error. Message should be different though. Don't use word "retry" there and explicitly say that specific argument must be changed.